File: Codeforces/1490A Dense Array.cpp
#include<bits/stdc++.h>
using namespace std;
int main()
{
    ios::sync_with_stdio(0);
    cin.tie(0), cout.tie(0);

    int tc, n, arr[55], temp, res;
    cin>>tc;
    while(tc--){
        cin>>n;
        res = 0;
        for(int i=0; i<n; i++) cin>>arr[i];
        for(int i=1, x, y; i<n; i++){
            x = min(arr[i-1], arr[i]);
            y = max(arr[i-1], arr[i]);
            if(x*2 < y){
                while(x*2 < y) res++, x *= 2;
            }
        }
        cout<<res<<endl;
    }

    return 0;
}

File: Codeforces/1490A Dense Array.py
tc = int(input())
lst = []
for cs in range(0, tc):
    res = 0
    n = int(input())
    line = input()
    num = list(map(int, line.split()))
    for i in range(1, n):
        x = min(num[i-1], num[i])
        y = max(num[i-1], num[i])
        while x*2 < y:
            x *= 2
            res = res + 1
    print(f"{res}")
